datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -452,6 +453,24 @@ private void validateRazEnablement(SdxClusterRequest sdxClusterRequest, Detailed
         }
     }
 
+    private void validateMediumDutySdxEnablement(SdxClusterRequest sdxClusterRequest, DetailedEnvironmentResponse environment) {
+        ValidationResultBuilder validationBuilder = new ValidationResultBuilder();
+        if (SdxClusterShape.MEDIUM_DUTY_HA.equals(sdxClusterRequest.getClusterShape())) {
+            boolean mediumDutySdxEntitlementEnabled = entitlementService.mediumDutySdxEnabled(environment.getCreator(),
+                    Crn.safeFromString(environment.getCreator()).getAccountId());
+            if (!mediumDutySdxEntitlementEnabled) {
+                validationBuilder.error("Provisioning a Medium Duty SDX shape is not enabled for this account.");
```

Comment:
Improve this error message to state what entitlement should be enabled.

"Provisioning a medium duty data lake cluster is not enabled for this account. Contact Cloudera support to enable CDP_MEDIUM_DUTY_SDX entitlement for the account"